### PR TITLE
Adopt Base's annotated types/functions as API

### PR DESCRIPTION
**Pull request by tecosaur**
This opens the door to shuffling parts currently implemented in Base into this stdlib without breaking any public APIs.

To try to help avoid people using `Base.Annot...` anyway, I've exported these bindings. I think this is reasonable given unlike in `Base` this library needs to be deliberately imported.

### CODE REVIEW

**Comment by tecosaur** [comment]
This change has been discussed with other maintainers on Slack (the API shift, but not the exporting), so I'm inclined to merge this soon if I can figure out how to make doctests happy.

**Review by topolarity — APPROVED** [review]
Looks good to me, thanks @tecosaur

**Comment by tecosaur** [comment]
This is held up by difficulty figuring out the Documenter side of things.

**Comment by tecosaur** [comment]
It's hard to see how I can make this work before https://github.com/JuliaDocs/Documenter.jl/issues/1781 is resolved :grimacing:.

Perhaps @mortenpi might have an idea?

**Comment by tecosaur** [comment]
In light of https://github.com/JuliaLang/julia/pull/55845 and the change Kristoffer's just pushed it seems like neither of us can easily make Documenter do what we want here, so let's just leave the documentation of the Base-defined types there with an appropriate notice.

I think all that needs to be done now is the `@ref StyledStrings.Annotated...` changes to be reverted back to `@ref Base.Annotated...` in this PR (it currently produces 404 links), and then we can merge it and stop being a 1.11 blocker.

**Pull request merged**
